internal/k8s/copy.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 Authors of Cilium
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+)
+
+const (
+	defaultReadFromByteCmd = "tail -c+%d %s"
+	defaultMaxTries        = 5
+)
+
+// CopyFromPod is to copy srcFile in a given pod to local destFile with defaultMaxTries.
+func (c *Client) CopyFromPod(ctx context.Context, namespace, pod, container string, srcFile, destFile string) error {
+	pipe := newPipe(&CopyOptions{
+		MaxTries: defaultMaxTries,
+		ReadFunc: readFromPod(ctx, c, namespace, pod, container, srcFile),
+	})
+
+	outFile, err := os.Create(destFile)
+	if err != nil {
+		return err
+	}
```

Comment:
This function is missing a `defer outFile.Close()` or similar.

Reply:
you are right, I forgot about this most of the time, and relying on linters. If I am not wrong, file is having finalizer i.e. fd will be garbage collected later, not idea in case of reuse of file descriptor.

Let me make a small update on this, thanks again.